napari/utils/migrations.py

```diff
@@ -1,13 +1,22 @@
 import inspect
 import warnings
 from functools import wraps
-from typing import Any, Callable, cast
+from typing import Any, Callable, NamedTuple, cast
 
 from napari.utils.translations import trans
 
 _UNSET = object()
 
 
+class DeprecatedProperty(NamedTuple):
```

Comment:
The fields of this named tuple are often used as a function parameters in the functions here. Do you think it makes sense to reuse this abstraction in those places instead of passing through positional parameters?